internal/repository/repository_test.go

```diff
@@ -0,0 +1,131 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package repository
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/suite"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/postgres"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const testTimeout = 300 * time.Second
+
+// This suite sets up a container running a postgres database
+type RepositorySuite struct {
+	suite.Suite
+	ctx      context.Context
+	cancel   context.CancelFunc
+	postgres *postgres.PostgresContainer
+	database *database
+}
+
+func (s *RepositorySuite) SetupSuite() {
+	s.ctx, s.cancel = context.WithTimeout(context.Background(), testTimeout)
```

Comment:
I get the safety preoccupation, but I'm not a fan of the hard-coded timeout. IMO such a timeout should be configured at the `go test` level (especially on the CI), as some machines will naturally take longer than others.

Reply:
Loved the setup/teardown, though 👌🏽

Reply:
I've just used the same configuration as used on the eth_util tests. I think, for consistency, we keep them the same.

Aside from that, there's a a discussion to be had that a 5 minute timeout is good actually, because if the test is taking that long, we really should raise an alarm as to something not being right.